crates/requirements-txt/src/lib.rs

```diff
@@ -436,10 +436,26 @@ impl RequirementsTxt {
         Ok(data)
     }
 
-    /// Merges other into self
+    /// Merge the data from a nested `requirements` file (`other`) into this one.
     pub fn update_from(&mut self, other: Self) {
-        self.requirements.extend(other.requirements);
-        self.constraints.extend(other.constraints);
+        let RequirementsTxt {
+            requirements,
+            constraints,
+            editables,
+            index_url,
+            extra_index_urls,
+            find_links,
+            no_index,
+        } = other;
+        self.requirements.extend(requirements);
+        self.constraints.extend(constraints);
+        self.editables.extend(editables);
+        if self.index_url.is_none() {
+            self.index_url = index_url;
+        }
+        self.extra_index_urls.extend(extra_index_urls);
+        self.find_links.extend(find_links);
+        self.no_index = self.no_index || no_index;
```

Comment:
We missed incorporating these extra fields when they were added to `RequirementsTxt`.